libyara/atoms.c

```diff
@@ -154,15 +155,13 @@ int yr_atoms_heuristic_quality(YR_ATOMS_CONFIG* config, YR_ATOM* atom)
         yr_bitmask_set(seen_bytes, atom->bytes[i]);
         unique_bytes++;
       }
+      literal_bytes++;
     }
   }
 
   // If all the bytes in the atom are equal and very common, let's penalize
   // it heavily.
-  if (unique_bytes == 1 && (yr_bitmask_is_set(seen_bytes, 0x00) ||
-                            yr_bitmask_is_set(seen_bytes, 0x20) ||
-                            yr_bitmask_is_set(seen_bytes, 0xCC) ||
-                            yr_bitmask_is_set(seen_bytes, 0xFF)))
+  if (unique_bytes == 1 && literal_bytes > 1)
```

Comment:
With this change it looks like an atom like 00 FF FF is considered better than 14 14 14 14, but I would say it's quite the contrary. I agree on taking into account repetitions of 0x90 in the heuristics (I've faced this issue myself), but treating all repetitions of any byte equally doesn't seem like the right approach. Not all repetitions are equally bad. 

What about penalizing the repetition of 0x90 as it's already done for 0x20, 0xCC and 0xFF?